fcrepo-http-api/src/main/java/org/fcrepo/http/api/services/HttpRdfService.java

```diff
@@ -117,6 +122,7 @@ public Model bodyToInternalModel(final String extResourceId, final InputStream s
 
         while (stmtIterator.hasNext()) {
             final Statement stmt = stmtIterator.nextStatement();
+            checkForDisallowedRdf(stmt);
```

Comment:
It seems like some of the functionality here is duplicating some checks in `AbstractService`:
https://github.com/fcrepo/fcrepo/blob/main/fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/services/AbstractService.java#L154
Were you thinking we would remove the check in AbstractService?

One advantage to where it is in AbstractService is that it is validating the model after a sparql update query has been applied to the model, in the case of PATCH. I don't think this method is called for PATCH currently?

Reply:
I don't know if this is a concern or not, but it might be unpredictable which error response you get from a request if the input contains both an interaction model change and a server managed triple. By checking for all the errors in the same loop (or not deferring the error thrown), whichever error is encountered first will be the result. Maybe that's fine, I was just concerned about that when looking at some tests, since the order of RDF is not generally guaranteed.

For what its worth, I stupidly hadn't noticed this PR yesterday and had started implementing checks for interaction model changes in my work on membership triples since so many of the Direct/Indirect container tests are about interaction model changes. I had added this method to `ReplacePropertiesServiceImpl`:
```
private void ensureInteractionModelUnchanged(final FedoraId fedoraId, final Model model,
            final String existingInteractionModel) {
        final var rdfResc = model.getResource(fedoraId.getFullId());
        final StmtIterator it = rdfResc.listProperties(RDF.type);
        while (it.hasNext()) {
            final Statement st = it.next();
            if (!st.getObject().isURIResource()) {
                continue;
            }
            final var objUri = st.getResource().getURI();

            if (INTERACTION_MODELS_FULL.contains(objUri) && !objUri.equals(existingInteractionModel)) {
                throw new InteractionModelViolationException("Changing the interaction model "
                        + existingInteractionModel + " to " + objUri + " is not allowed!");
            }
        }
    }
```
I was running this before `checkForSmtsLdpTypes`. On the other hand adding more iterations through the model will add up.

 but wanted to make sure that if the request was changing the interaction model, that we would get the error for that over an error for

Reply:
You are right about some duplication at the AbstractService layer. I had forgotten about that. As we are already looping through the entire incoming RDF/Sparql-update at the Http layer perhaps removing those checks from the lower level would be better?

This does check PATCH requests as we also loop through them to translate any external URIs to internal IDs.

As for storing up exceptions...I'm a believer in here is your first problem, come back later. But also I am wondering is if you try to change a server managed triple and change the interaction model in a single request. Do you expect a 409 or 400?

Reply:
Seems like we either need to define precedence for exception types (maybe prioritize specificity?), or leave it as a free for all. Flapping tests seem possible without any sort of precedence, but that doesn't seem to have been a problem so far. Not sure if that's a function of jena not changing significantly, the order in which checks happen in Fedora 4/5, or how the tests are written.